target/linux/ath79/image/nand.mk

```diff
@@ -77,6 +90,43 @@ define Device/domywifi_dw33d
 endef
 TARGET_DEVICES += domywifi_dw33d
 
+define Device/dongwon_dw02-412h
+  SOC := qca9557
+  DEVICE_VENDOR := Dongwon T&I
+  DEVICE_MODEL := DW02-412H
+  DEVICE_ALT0_VENDOR := T&Ie
+  DEVICE_ALT0_MODEL := DW02-412H
+  DEVICE_ALT1_VENDOR := KT
+  DEVICE_ALT1_MODEL := GiGA WiFi home
```

Comment:
The DEVICE_ALT* strings will appear on menuconfig and it will confuse users if the specific model name is not shown there. So I would add `DEVICE_ALT1_VARIANT := (DW02-412H)` here.

And...why did you add `DEVICE_ALT0_VENDOR := T&Ie` here?

Reply:
I did it by referring to an [example of this device](https://github.com/openwrt/openwrt/blob/c58005a9431b16107d8c6398b9095a5b3bcfcb25/target/linux/ath79/image/generic.mk#L220) which included both old and new company name...
I have a device that was made after the company name was changed to T&Ie.
However, I am considering deleting this vendor because I think it will make the situation very confusing by adding this.

Reply:
> The DEVICE_ALT* strings will appear on menuconfig and it will confuse users if the specific model name is not shown there. So I would add `DEVICE_ALT1_VARIANT := (DW02-412H)` here.

![variant added](https://user-images.githubusercontent.com/79572354/109534351-9a10b480-7afe-11eb-9108-97c5b53d67e3.PNG)
![variant removed](https://user-images.githubusercontent.com/79572354/109534381-a1d05900-7afe-11eb-9807-28b33bab8362.PNG)

I think it's a good suggestion, but this seems to make the target profile too long.

Reply:
> I did it by referring to an example of this device which included both old and new company name...

I would do that only if there's a device with a label "T&Ie DW02-412H".

> I think it's a good suggestion, but this seems to make the target profile too long.

Oh, I didn't know that the original name is appended there... My bad.
BTW menuconfig is notorious for its small menu window... nconfig is better in that perspective.
That name will also appear at the [firmware-selector](https://firmware-selector.openwrt.org/), so I still think we should put DEVICE_ALTx_VARIANT here.

Reply:
Okay. I'll remove `T&Ie` vendor and add `DEVICE_ALT0_VARIANT := (DW02-412H)`.

Reply:
Oh, and don't forget to append "{64,128}M" to it. (Wow, it really gets longer and longer...)